src/osd/scheduler/mClockScheduler.cc

```diff
@@ -123,9 +123,7 @@ OpSchedulerItem mClockScheduler::dequeue()
   } else {
     mclock_queue_t::PullReq result = scheduler.pull_request();
     if (result.is_future()) {
-      ceph_assert(
-	0 == "Not implemented, user would have to be able to be woken up");
-      return std::move(*(OpSchedulerItem*)nullptr);
+      return std::move(OpSchedulerItem(result.getTime()));
```

Comment:
This is pretty confusing (what if someone calls enqueue on a "future" OpQueueitem?).  Rather than embedding this entire other concept into OpQueueItem, can we change dequeue() to return a std::variant or pass a pointer to a ceph::real_time out param?

Reply:
I changed dequeue() to return a std::variant as suggested. Please take a look.

Reply:
I have changed dequeue() to return std::variant as suggested. Please take a look.